engine/core/src/main/java/org/datacleaner/job/builder/AnalysisJobBuilder.java

```diff
@@ -1205,4 +1219,37 @@ public Boolean eval(InputColumn<?> inputColumn) {
 
         return result;
     }
+
+    public AnalysisJobBuilder getTopLevelJobBuilder(){
+        AnalysisJobBuilder builder = this;
+        AnalysisJobBuilder tempBuilder = builder._parentBuilder;
+        while(tempBuilder != null){
+            builder = tempBuilder;
+            tempBuilder = builder._parentBuilder;
+        }
+
+        return builder;
+    }
+
+    private List<AnalysisJobBuilder> getChildren() {
+        List<AnalysisJobBuilder> childJobBuilders = new ArrayList<>();
+
+        for (ComponentBuilder componentBuilder : getComponentBuilders()) {
+            for (OutputDataStream outputDataStream : componentBuilder.getOutputDataStreams()) {
+                childJobBuilders.add(componentBuilder.getOutputDataStreamJobBuilder(outputDataStream));
+            }
+        }
+
+        return childJobBuilders;
+    }
+
+    public List<AnalysisJobBuilder> getDescendants() {
```

Comment:
I actually loved finding a name with no annoying repeat of the class name... But will add if it causes confusion.